Generator.md

```diff
@@ -0,0 +1,356 @@
+        }
+
+        internal static IServiceCollection Register(this IServiceCollection serviceCollection)
+        {
+            serviceCollection.AddSingleton<IService, Service>();
+            serviceCollection.AddSingleton(sp => new CustomerServiceFactory().Factory(sp));
+            serviceCollection.AddSingleton(typeof(IDoWork<>),typeof(GenericDoWork<>)); // <=== Here
+            // Of Collection code
+            serviceCollection.AddSingleton<IPipeline, Pipeline1>();
+            serviceCollection.AddSingleton<IPipeline, Pipeline2>();
+            serviceCollection.AddSingleton<IPipeline, Pipeline3>();
+
+            return serviceCollection;
+        }
+    }
+```
+
+`ReplaceService = true` is a parameter that can be specified on the register property when it becomes necessary to replace a service with a mock version or a new implementation.
+
+``` csharp
+using Agoda.IoC.Generator.Abstractions;
+
+namespace Agoda.Example;
+
+// Old
+public interface IService { }
+[RegisterSingleton]
+public class Service : IService { }
+
+
+//New
+[RegisterSingleton(ReplaceService = true)]
+public class NewService : IService { }
+```
+
+Project analyzers contain the complete source code.
+``` csharp
+// <auto-generated>
+//     Automatically generated by Agoda.IoC.Generator.
+//     Changes made to this file may be lost and may cause undesirable behavior.
+// </auto-generated>
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Agoda.Example;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class AgodaIoCGeneratorInjectServiceCollectionExtension
+    {
+        public static IServiceCollection RegisterFromAgodaExample(this IServiceCollection serviceCollection)
+        {
+            return Register(serviceCollection);
+        }
+
+        internal static IServiceCollection Register(this IServiceCollection serviceCollection)
+        {
+            serviceCollection.AddSingleton<IService, Service>();
+            serviceCollection.Replace(new ServiceDescriptor(typeof(NewService), ServiceLifetime.Singleton)); // <=== HERE
+            serviceCollection.AddSingleton(sp => new CustomerServiceFactory().Factory(sp));
+            serviceCollection.AddSingleton(typeof(IDoWork<>),typeof(GenericDoWork<>));
+            // Of Collection code
+            serviceCollection.AddSingleton<IPipeline, Pipeline1>();
+            serviceCollection.AddSingleton<IPipeline, Pipeline2>();
+            serviceCollection.AddSingleton<IPipeline, Pipeline3>();
+
+            return serviceCollection;
+        }
+    }
+}
+```
+
+
+# Final step
+
+You just need to call extension method in entry project like this
+
+``` csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+builder.Services.RegisterFromAgodaExample(); // Here 😃 
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
```

Comment:
I would remove all these other lines like swagger and controllers. Just have the container build and the commands needed for this library. Or it may confuse some people

Reply:
removed